deployment/ccip/changeset/deployer_group.go

```diff
@@ -0,0 +1,165 @@
+package changeset
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+)
+
+type DeployerGroup struct {
+	e            deployment.Environment
+	state        CCIPOnChainState
+	mcmConfig    *MCMSConfig
+	transactions map[uint64][]*types.Transaction
+}
+
+// DeployerGroup is an abstraction that lets developers write their changeset
+// without needing to know if it's executed using a DeployerKey or an MCMS proposal.
+//
+// Example usage:
+//
+//	deployerGroup := NewDeployerGroup(e, state, mcmConfig)
+//	selector := 0
+//	# Get the right deployer key for the chain
+//	deployer := deployerGroup.getDeployer(selector)
+//	state.Chains[selector].RMNRemote.Curse()
+//	# Execute the transaction or create the proposal
+//	deployerGroup.enact("Curse RMNRemote")
+func NewDeployerGroup(e deployment.Environment, state CCIPOnChainState, mcmConfig *MCMSConfig) *DeployerGroup {
+	return &DeployerGroup{
+		e:            e,
+		mcmConfig:    mcmConfig,
+		state:        state,
+		transactions: make(map[uint64][]*types.Transaction),
+	}
+}
+
+func (d *DeployerGroup) getDeployer(chain uint64) *bind.TransactOpts {
+	txOpts := d.e.Chains[chain].DeployerKey
+	if d.mcmConfig != nil {
+		txOpts = deployment.SimTransactOpts()
+	}
+	sim := &bind.TransactOpts{
+		From:       txOpts.From,
+		Signer:     txOpts.Signer,
+		GasLimit:   txOpts.GasLimit,
```

Comment:
Not 100% sure what these values would be in `DeployerKey` but if this is non-zero the gethwrappers won't estimate gas, see e.g transactions https://github.com/ethereum/go-ethereum/blob/c0882429f032da58620bcaa610007939aa7e0adb/accounts/abi/bind/base.go#L316

Reply:
I see but I don't think we would be modifying the original behavior from the deployerKey here no?